lib/model.js

```diff
@@ -972,12 +972,35 @@ class Model extends Entity {
             }
 
             // ...Wrong format
-            if (!skip && schemaHasProperty && !isValueEmpty &&
-                    {}.hasOwnProperty.call(schema.paths[k], 'validate') &&
-                    !validator[schema.paths[k].validate](propertyValue)) {
-                errors[k] = new GstoreError.ValidatorError({
-                    message: `Wrong format for property { ${k} }`,
-                });
+            if (
+                !skip && schemaHasProperty && !isValueEmpty &&
+                {}.hasOwnProperty.call(schema.paths[k], 'validate')
+            ) {
+                /**
+                 * By default, the args only contain the value being validated
+                 * and the string, matching the simple case.
+                 */
+                let validationArgs = [propertyValue];
+                let validationRule = schema.paths[k].validate;
+
+                /**
+                 * If the validate property is an object, then it's assumed that
+                 * it contains the 'rule' property, which will be the new
+                 * validationRule's value.
+                 * If the 'args' prop was passed then we concat them to
+                 * 'validationArgs'.
+                 */
+                if (typeof schema.paths[k].validate === 'object') {
+                    validationRule = schema.paths[k].validate.rule;
+                    validationArgs = validationArgs.concat(schema.paths[k].validate.args || []);
+                }
+
+                /** Then it gets easier to use 'apply' on both simple and complex cases */
+                if (!validator[validationRule].apply(validator, validationArgs)) {
+                    errors[k] = new GstoreError.ValidatorError({
+                        message: `Wrong format for property { ${k} }`,
+                    });
+                }
```

Comment:
This is perfect. Could you maybe just change line 993 with `if (typeof validationRule === 'object') ` just for readibility?